cbmc/proofs/DeserializeSuback/DeserializeSuback_harness.c

```diff
@@ -0,0 +1,23 @@
+#include AWS_IOT_CONFIG_FILE
+#include "private/aws_iot_mqtt_internal.h"
+
+#include <stdlib.h>
+
+void harness()
+{
+  AwsIotMqttConnection_t mqttConnection;
+  size_t dataLength;
+  uint8_t *pSubackStart = malloc(sizeof(uint8_t) * dataLength);
+  uint16_t PacketIdentifier;
+  size_t BytesProcessed;
+
+  __CPROVER_assume(dataLength <= BUFFER_SIZE);
```

Comment:
Why do we need this for this harness only? None of the other harnesses that have a `dataLength` need this assumption.

Are we proving that this assumption is true in some other harness? If so, please write a comment saying which one.

Reply:
All other proofs malloc an unconstrained buffer of an unconstrained size.  This proof mallocs an unconstrained buffer of an unconstrained *but bounded* size because this is the only parser that actually iterates over bytes in the buffer (meaning there is a loop in the code requiring a concrete bound for unwinding).  All others know where to look in the buffer and just do the required pointer arithmetic to get what they need.